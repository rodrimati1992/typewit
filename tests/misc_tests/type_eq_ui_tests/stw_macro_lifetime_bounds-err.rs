use typewit::MakeTypeWitness;

typewit::simple_type_witness!{
    enum LifetimeBounds['a, 'b: 'a, 'c: 'b + 'a, 'd: 'a + 'static] {
        U8 = (&'a (), &'b (), &'c (), &'d ()) 
    }
}

fn _foo<'a, 'b, 'c, 'd>() {
    let _: LifetimeBounds<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
}

fn main(){}