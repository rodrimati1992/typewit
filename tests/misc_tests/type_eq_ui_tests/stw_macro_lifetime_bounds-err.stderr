error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:16:12
   |
15 | fn _foo<'a, 'b, 'c, 'd>() {
   |         --  -- lifetime `'b` defined here
   |         |
   |         lifetime `'a` defined here
16 |     let _: LifetimeBounds<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'b` must outlive `'a`
   |
   = help: consider adding the following bound: `'b: 'a`

error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:16:12
   |
15 | fn _foo<'a, 'b, 'c, 'd>() {
   |         --      -- lifetime `'c` defined here
   |         |
   |         lifetime `'a` defined here
16 |     let _: LifetimeBounds<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'c` must outlive `'a`
   |
   = help: consider adding the following bound: `'c: 'a`

error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:16:12
   |
15 | fn _foo<'a, 'b, 'c, 'd>() {
   |                     -- lifetime `'d` defined here
16 |     let _: LifetimeBounds<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'d` must outlive `'static`

help: the following changes may resolve your lifetime errors
  |
  = help: add bound `'b: 'a`
  = help: add bound `'c: 'a`
  = help: replace `'d` with `'static`

error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:20:12
   |
19 | fn _bar<'a, 'b, 'c, 'd>() {
   |         --  -- lifetime `'b` defined here
   |         |
   |         lifetime `'a` defined here
20 |     let _: LifetimeBoundsAngBrack<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'b` must outlive `'a`
   |
   = help: consider adding the following bound: `'b: 'a`

error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:20:12
   |
19 | fn _bar<'a, 'b, 'c, 'd>() {
   |         --      -- lifetime `'c` defined here
   |         |
   |         lifetime `'a` defined here
20 |     let _: LifetimeBoundsAngBrack<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'c` must outlive `'a`
   |
   = help: consider adding the following bound: `'c: 'a`

error: lifetime may not live long enough
  --> tests/misc_tests/type_eq_ui_tests/stw_macro_lifetime_bounds-err.rs:20:12
   |
19 | fn _bar<'a, 'b, 'c, 'd>() {
   |                     -- lifetime `'d` defined here
20 |     let _: LifetimeBoundsAngBrack<'a, 'b, 'c, 'd, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'d` must outlive `'static`
